### DISCUSSION TITLE
Non-customizable horizon leveling

### DISCUSSION THREAD

**Discussion by czukowski** [gopro/labs#589] [Feature Requests]
I've seen the other thread but my question is whether it would be possible to just enable it on Wide lens and let it do its thing?

In my experience, I only ever need some very small adjustments. If a frame clipped, I can deal with it in post processing. I've already tried Max Lens 2.0 but unfortunately, I'm not happy with the results so far.

**Comment by dnewman-gpsw** [comment]
HL with Wide, but have the edges clip?  Why not do horizon levelling in post?

**Reply by czukowski** [reply]
That's what I do now but it's proven very tedious because the tilt is not constant, it's usually somewhere within 2 to 3°, sometimes more during some sharp turns.

For the most part, the scenes are not very dynamic (camera movement wise) so the background scenery stays in the frame for a while and often times there is a horizon over water visible and then even 1° tilt is very apparent. Other times, like in a forest, there may be no good point of reference to see the tilt, but sometimes it still gives a feeling that something's off, it's just harder to tell what it is. That's why I'm trying to fix it, and these are some lengthy clips. I do apply some lens correction in post that gives enough space to tolerate tilts up to 10° degrees or so.

**Reply by dnewman-gpsw** [reply]
ReelSteady is making these errors?  It should be using the same computed gravity vector that is stored in metadata, so in-camera leveling should be the same a post.

**Reply by czukowski** [reply]
No, I don't use ReelSteady, to my shame, I've only recently discovered that it existed 😅 I may try to see how it copes with my use case but I understand it that it's a standalone software, and I would have to add a new transcoding step before continuing with my usual workflow? From what I can tell, it can't be used as an OpenFX plug-in or export the calculated motion data in some way (for me to somehow script into my workflow). I did try Gyroflow that also can use gravity vectors from metadata as well as stabilization based on optical flow, it does help, but still far from perfect. That's why I'm looking for new approaches. GoPro has already horizon leveling built-in, it's just too limited. Of course, clipped edges are not good on a consumer product, but such an option for Labs firmware would be prefect for enthusiasts, provided it's feasible to do.

**Reply by dnewman-gpsw** [reply]
I was talking of the ReelSteady option built into GoPro Player.  Just turn of Hypersmooth in the camera, ReelSteady in Player offer horizon leveling.

**Reply by czukowski** [reply]
I understand, I'll try and see how it looks like, but I'm doing much more editing than just leveling the horizon, so another thing I'll have to solve is how to take that leveled output and work with it.

**Reply by czukowski** [reply]
I'm still waiting for an opportunity to go and shoot a clip with Hypersmooth off like you suggested, but below is a screenshot from GoPro Player with lens correction, shot with Max Lens Mod 2.0 and Horizon Lock enabled (there's also an on-screen ruler for a visual aid). I guess, if it looks like that, then I would get the same thing even if I were able to turn on Horizon Leveling without Max Lens?

![Screenshot 2023-10-08 121437](https://github.com/gopro/labs/assets/186792/e2f573a1-5655-4f12-80b1-0f2382394dc1)

**Reply by dnewman-gpsw** [reply]
Correct, all use the same data.

**Reply by czukowski** [reply]
I thought I was getting these small deviances because I normally record _without_ Horizon Level/Lock but I see it's more or less the same so I guess I'm stuck with keyframing... 🤔

**Reply by dnewman-gpsw** [reply]
Fusion for gravity calculation is very hard.  The accelerometer measures both camera movement and forces of gravity, disambiguating those is what can lead to small horizon errors over time.
